modules/client-handler/src/main/java/org/apache/ignite/client/handler/requests/jdbc/JdbcMetadataCatalog.java

```diff
@@ -211,30 +279,31 @@ public CompletableFuture<Collection<String>> getSchemasMeta(String schemaNamePtr
      * @param tbl Table.
      * @return Jdbc primary key metadata.
      */
-    private JdbcPrimaryKeyMeta createPrimaryKeyMeta(CatalogTableDescriptor tbl) {
+    private static JdbcPrimaryKeyMeta createPrimaryKeyMeta(String schemaName, CatalogTableDescriptor tbl) {
         String keyName = PK + tbl.name();
 
         List<String> keyColNames = List.copyOf(tbl.primaryKeyColumns());
 
-        return new JdbcPrimaryKeyMeta(SqlCommon.DEFAULT_SCHEMA_NAME, tbl.name(), keyName, keyColNames);
+        return new JdbcPrimaryKeyMeta(schemaName, tbl.name(), keyName, keyColNames);
     }
 
     /**
      * Creates column metadata from column and table name.
      *
+     * @param schemaName Schema name.
      * @param tblName Table name.
-     * @param col     Column.
+     * @param col Column descriptor.
      * @return Column metadata.
      */
-    private JdbcColumnMeta createColumnMeta(String tblName, Column col) {
-        NativeType colType = col.type();
+    private static JdbcColumnMeta createColumnMeta(String schemaName, String tblName, CatalogTableColumnDescriptor col) {
+        NativeType colType = TypeUtils.columnType2NativeType(col.type(), col.precision(), col.scale(), col.length());
```

Comment:
```suggestion
```
Let's use precition/scale from column descriptor directly.

Reply:
This method is used to correctly convert length to precision for VARLEN types